code/modules/wiremod/components/hud/bar_overlay.dm

```diff
@@ -0,0 +1,62 @@
+/**
+ * # Bar Overlay Component
+ *
+ * Basically an advanced verion of object overlay component that shows a horizontal/vertical bar.
+ * Requires a BCI shell.
+ */
+
+#define BAR_OVERLAY_LIMIT 10
+
+/obj/item/circuit_component/object_overlay/bar
+	display_name = "Bar Overlay"
+	display_desc = "Requires a BCI shell. A component that shows a bar overlay ontop of an object from a range of 0 to 100."
+
+	var/datum/port/input/bar_number
+
+/obj/item/circuit_component/object_overlay/bar/Initialize()
+	. = ..()
+	bar_number = add_input_port("Number", PORT_TYPE_NUMBER)
+
+	set_option("Vertical")
```

Comment:
If you want vertical to be the default, have it first in the component_options_bar list.